met-web/src/routes/AuthGate.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { useAppSelector } from 'hooks';
+import { useLocation, Navigate, Outlet } from 'react-router-dom';
+
+const AuthGate = ({ allowedRoles }: { allowedRoles: string[] }) => {
+    const permissions = useAppSelector((state) => state.user.roles);
+    const location = useLocation();
+
+    const scopesMap: { [scope: string]: boolean } = {};
+    allowedRoles.forEach((scope) => {
+        scopesMap[scope] = true;
+    });
+
+    return permissions.some((permission) => scopesMap[permission]) ? (
+        <Outlet />
+    ) : (
+        <Navigate to="/unauthorized" state={{ from: location }} replace />
+    );
+};
+
```

Comment:
I think some comment could be helpful here since we use scope and roles interchangeably..
I think scope=role(i mean the variables) here rite?

Reply:
may be bring a new variable ...
```
const authorized = = permissions.some((permission) => scopesMap[permission])
return authorized ? (
  <Outlet />
) : (
  <Navigate to="/unauthorized" state={{ from: location }} replace />
);
```

Reply:
cool will handle in subsequent PR